src/zenml/integrations/gcp/model_registries/vertex_model_registry.py

```diff
@@ -0,0 +1,309 @@
+        if name:
+            filter_expr = filter_expr + f' AND  display_name="{name}"'
+        if metadata:
+            for key, value in metadata.items():
+                filter_expr = filter_expr + f' AND  labels.{key}="{value}"'
+        try:
+            models = aiplatform.Model.list(filter=filter_expr)
+            return [
+                RegisteredModel(
+                    name=model.display_name,
+                    description=model.description,
+                    metadata=model.labels,
+                )
+                for model in models
+            ]
+        except Exception as e:
+            raise RuntimeError(f"Failed to list models: {str(e)}")
+
+    def register_model_version(
+        self,
+        name: str,
+        version: Optional[str] = None,
+        model_source_uri: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[ModelRegistryModelMetadata] = None,
+        **kwargs: Any,
+    ) -> RegistryModelVersion:
+        """Register a model version to the Vertex AI model registry."""
+        self.setup_aiplatform()
+        metadata_dict = metadata.model_dump() if metadata else {}
+        serving_container_image_uri = metadata_dict.get(
+            "serving_container_image_uri",
+            None
+            or "europe-docker.pkg.dev/vertex-ai/prediction/sklearn-cpu.1-3:latest",
+        )
+        is_default_version = metadata_dict.get("is_default_version", False)
+        metadata_dict["managed_by"] = "zenml"
+        try:
+            version_info = aiplatform.Model.upload(
+                artifact_uri=model_source_uri,
+                display_name=f"{name}_{version}",
+                serving_container_image_uri=serving_container_image_uri,
+                description=description,
+                is_default_version=is_default_version,
+                labels=metadata_dict,
+            )
+            return RegistryModelVersion(
+                version=version_info.version_id,
+                model_source_uri=version_info.resource_name,
+                model_format="Custom",  # Vertex AI doesn't provide this info directly
+                registered_model=self.get_model(version_info.name),
+                description=description,
+                created_at=version_info.create_time,
+                last_updated_at=version_info.update_time,
+                stage=ModelVersionStage.NONE,  # Vertex AI doesn't have built-in stages
+                metadata=metadata,
+            )
+        except Exception as e:
+            raise RuntimeError(f"Failed to register model version: {str(e)}")
+
+    def delete_model_version(
+        self,
+        name: str,
+        version: str,
+    ) -> None:
+        """Delete a model version from the Vertex AI model registry."""
+        self.setup_aiplatform()
+        try:
+            model_version = aiplatform.ModelVersion(
+                model_name=f"{name}@{version}"
```

Comment:
Above for the display name we use `name_version`, are you sure this is the correct name to delete? Where to we specify it?

Reply:
After reading the VertexAI docs, it was apparent to me that they actually support this type of syntax, where `X@Y` really means `version X of model Y`. However, this only works with model IDs, not with the model display name.